runtime/reconcilers/metrics_view.go

```diff
@@ -76,12 +77,16 @@ func (r *MetricsViewReconciler) Reconcile(ctx context.Context, n *runtimev1.Reso
 	if validateErr == nil {
 		validateErr = validateResult.Error()
 	}
+
 	if ctx.Err() != nil {
 		return runtime.ReconcileResult{Err: errors.Join(validateErr, ctx.Err())}
 	}
 	if validateErr == nil {
 		mv.State.ValidSpec = mv.Spec
 	} else {
+		if os.Getenv("DEBUG") == "1" {
+			fmt.Println("Reconciler error", validateErr.Error())
+		}
```

Comment:
Remove

Reply:
It took me several hours of debugging without those log messages, I don't understand. Reconciler and validation errors aren't shown if tests fail

Reply:
You just need to uncomment this line to get reconciler logs from tests: https://github.com/rilldata/rill/blob/1983fb1194d9f358eb5321534b15a4a824b0910c/runtime/testruntime/testruntime.go#L68

Reply:
Unless it's a very special circumstance, we should only read ENV variables at the entrypoint (e.g. CLI command or test function) and then pass config explicitly to internal packages, and we should use the Zap logger consistently instead of `fmt.Println`.

Reply:
`Unless it's a very special circumstance, we should only read ENV variables at the entrypoint (e.g. CLI command or test function)` I doubt that, in terms of speed of development I would rather add everywhere 
```
if os.Getenv("DEBUG") == "1") {
fmt.Println(sql)
}
```
and forgot about adding/removing and checking `fmt.Println(sql)` all the time.

Reply:
No, I can keep DEBUG=1 in config on my local machine.